### PR TITLE
[TSLint] Fix no typedef issues in some of our unit tests

### PR DESCRIPTION
**Pull request by Shobhit1**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000 - N/A
- [ ] Added relevant unit test for your changes. (`npm run test`) - N/A
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage` - N/A
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes. - N/A

#### Description of changes

Continuation of changes to add return typedef information on all our functions. In this PR, a lot of tests are covered.

### CODE REVIEW

File: src/tests/unit/tests/injected/analyzer-controller.test.ts [hunk 1]
```diff
@@ -243,10 +243,10 @@ describe('AnalyzerControllerTests', () => {
 });
 
 class AnalyzerStub implements IAnalyzer<any> {
-    public analyze() {
+    public analyze(): Error {
         throw new Error('Method not implemented.');
     }
-    public teardown() {
+    public teardown(): Error {
```
Comment (smoralesd) [inline src/tests/unit/tests/injected/analyzer-controller.test.ts]: shouldn't this be `void`?

Comment (Shobhit1) [reply src/tests/unit/tests/injected/analyzer-controller.test.ts]: It doesn't matter for this case, where `throw new Error()` is the only command in the function and I am fine with this function returning `Error`. Until you feel very strongly about me returning `void` here, I guess we can keep this as `Error`

Comment (Shobhit1) [reply src/tests/unit/tests/injected/analyzer-controller.test.ts]: changed it to "void"

File: src/tests/unit/tests/injected/visualization/drawer.test.ts [hunk 2]
```diff
@@ -1153,7 +1153,7 @@ describe('Drawer', () => {
                 throw new Error('Not implemented');
             }
 
-            public getDialogRenderer() {
+            public getDialogRenderer(): any {
```
Comment (dbjorge) [inline src/tests/unit/tests/injected/visualization/drawer.test.ts]: > any [](start = 39, length = 4)

Can you instead update `Formatter` to use `DialogRenderer?` and match that type here?

File: src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts [hunk 3]
```diff
@@ -996,7 +996,7 @@ describe('SVGDrawerV2Tests', () => {
             });
     }
 
-    function setupFilterFactoryDefault(dom) {
+    function setupFilterFactoryDefault(dom): void {
```
Comment (dbjorge) [inline src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts]: >dom [](start = 39, length = 3)

Should the parameter also have a type? I'm surprised that the line passes the lint check without that

**Comment by dbjorge** [comment]
>     function setupSVGshapeFactoryDefault(dom): void {

Should this param have a type?

---
Refers to: src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts:1010 in 9b3f69f. [](commit_id = 9b3f69fb3471fb2f60b1bbb740556bbc8eba9fef, deletion_comment = False)

File: src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts (hunk 3 above)
Comment (Shobhit1) [reply src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts]: There is no check as of yet to do that and personally I have not been touching those, So I would rather take that as a separate task with different specific PRs for those ones. 👍

Comment (dbjorge) [reply src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts]: Do you mean "we haven't enabled such a check"? I think the check exists (as the `parameter` argument to the `typedef` rule)

Comment (dbjorge) [reply src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts]: (regardless, fine with doing as a separate PR)

Comment (Shobhit1) [reply src/tests/unit/tests/injected/visualization/svg-drawer-v2.test.ts]: Discussed offline, this should be discussed in our next engineering discussion meeting and hopefully come up with some solution for this 😄

File: src/tests/unit/tests/injected/visualization/drawer.test.ts (hunk 2 above)
Comment (Shobhit1) [reply src/tests/unit/tests/injected/visualization/drawer.test.ts]: Done.

**Pull request merged**